src/NuGet.Core/NuGet.Packaging/Rules/IconMaxFilesizeRule.cs

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.IO;
+using NuGet.Common;
+using System.CodeDom;
+using System.Globalization;
+using System.Runtime.CompilerServices;
+
+namespace NuGet.Packaging.Rules
+{
+    public class IconMaxFilesizeRule : IPackageRule
+    {
+        private const int MaxIconFilzeSize = 1024 * 1024;
+
+        private const int BufferSize = 1024 * 1024;
+
+        public string MessageFormat { get; }
+
+        public IconMaxFilesizeRule(string messageFormat)
+        {
+            MessageFormat = messageFormat ?? throw new ArgumentNullException(nameof(messageFormat));
+        }
+
+        public IEnumerable<PackagingLogMessage> Validate(PackageArchiveReader builder)
+        {
+            var reader = builder?.NuspecReader;
+
+            var path = reader?.GetIcon();
+
+            var issues = new List<PackagingLogMessage>();
+
+            if (!string.IsNullOrEmpty(path))
```

Comment:
I don't think this can happen. 
If it does it's a bug in our implementation.

Reply:
When there's no <icon /> file entry, NuspecReader.GetIcon() will return an empty string. In that case, it is not necessary to validate the icon entry.